src/Merge/ExtraPackage.php

```diff
@@ -125,13 +127,20 @@ protected function loadPackage($json)
     /**
      * Merge this package into a RootPackage
      *
-     * @param RootPackage $root
+     * @param RootPackageInterface $top
      * @param PluginState $state
      */
-    public function mergeInto(RootPackage $root, PluginState $state)
+    public function mergeInto(RootPackageInterface $top, PluginState $state)
     {
-        $this->mergeRequires($root, $state);
-        $this->mergeDevRequires($root, $state);
+        if ($top instanceof RootAliasPackage)
```

Comment:
Yes, it does and that's the problem - it unwraps alias too early. Note the call to `mergeRequires` below needs $top, which wouldn't be available if alias is unwrapped in PluginState.

But this code can be replaced by `$root = $state->getRootPackage();`. I wasn't sure what's the best way. I just think that the current code is easier to understand as it shows clear connection between $top and $root.